qa/tasks/cephfs/test_nfs.py

```diff
@@ -487,6 +488,81 @@ def update_export(self, cluster_id, path, pseudo, fs_name):
+            f"- /var/log/ceph/{fsid}:/var/log/ceph:z", f'- "/var/log/ceph/{fsid}:/var/log/ceph:z"')
+        log.debug(f"debug enabled ganesha spec: {debug_enabled_ganesha_spec}")
+        self.ctx.cluster.run(
+            args=['ceph', 'orch', 'apply', '-i', '-'],
+            stdin=debug_enabled_ganesha_spec)
+
+        # wait for redeploy otherwise the below config changes would reset
+        time.sleep(5)
+
+        # add client debug to /var/lib/ceph/$fsid/$ganesha_daemon/config
+        ganesha_daemon = ((self._orch_cmd("ps", "--daemon-type", "nfs")).split("\n")[1].split(' ')[0]).strip()
+        GANESHA_CONF_FILE_PATH = f"/var/lib/ceph/{fsid}/{ganesha_daemon}/config"
+
+        original_ganesha_conf = (self.ctx.cluster.run(args=["sudo", "cat", GANESHA_CONF_FILE_PATH],
+                                                      stdout=StringIO(),
+                                                      stderr=StringIO()))[0].stdout.getvalue().strip()
+        if "[client]" not in original_ganesha_conf:
+            s = f"[client]\n\tdebug client = 20\n\tlog file = /var/log/ceph/ceph-client.nfs.{cluster_name}.log"
+            self._sys_cmd(["echo", Raw(f'"{s}"'), Raw("|"), "sudo", "tee", Raw("-a"), GANESHA_CONF_FILE_PATH])
+
+        self._orch_cmd("restart", f"nfs.{cluster_name}")
+
+        ganesha_conf_debug_enabled = (self.ctx.cluster.run(args=["sudo", "cat", GANESHA_CONF_FILE_PATH],
+                                                           stdout=StringIO(),
+                                                           stderr=StringIO()))[0].stdout.getvalue().strip()
+        self.assertIn("[client]", ganesha_conf_debug_enabled)
+        self.assertIn("debug client = 20", ganesha_conf_debug_enabled)
+        self.assertIn(f"log file = /var/log/ceph/ceph-client.nfs.{cluster_name}.log", ganesha_conf_debug_enabled)
+
+        time.sleep(5)
+
+        libcephfs_log = (self.ctx.cluster.run(args=["sudo", "cat",
+                                                    f"/var/log/ceph/{fsid}/ceph-client.nfs.{cluster_name}.log",
+                                                    Raw("|"), "tail", "-n", "2"],
+                                              check_status=True,
+                                              stdout=StringIO(),
+                                              stderr=StringIO()))[0].stdout.getvalue().strip()
+        self.assertGreater(len(libcephfs_log), 0, "empty log file")
+
+        return original_ganesha_spec, GANESHA_CONF_FILE_PATH, original_ganesha_conf
+
+    def disable_libcephfs_logging(self, cluster_name, ganesha_spec, conf_path, ganesha_conf):
+        self.ctx.cluster.run(
+            args=['ceph', 'orch', 'apply', '-i', '-'],
+            stdin=ganesha_spec)
+        
+        # ceph orch apply should reset the config, wait for 5 secs if it still exists
+        # then move ahead with manually removing it
+        time.sleep(5)
```

Comment:
`wait_until_true` with ~60s timeout would be a better structure rather than an arbitrary sleep duration. No one will want test failures where `5.1` seconds would have worked instead of `5`.

Reply:
the redeployment is usually within ~1-2 sec, i've intentionally kept it 5 to have that room for any lag (though I agree we might not know when it overshoots). I can have the structure in place but thought to keep it simple.

Reply:
I spent a lot of time to figure out a way to mimic `assert_cluster_logs` style checking for redeployment but I couldn't fine an easy and simple approach than checking PID before and after the daemon redeployment. PTAL https://github.com/ceph/ceph/pull/61547/commits/c4818f5856008d8db59c514cc77fbf5382bc4903